test/helper.rb

```diff
@@ -63,36 +70,61 @@ def get_adapter_name
 class FauxWriter < Datadog::Writer
   def initialize
     super(transport: FauxTransport.new(HOSTNAME, PORT))
+    @mutex = Mutex.new
 
     # easy access to registered components
     @spans = []
     @services = {}
   end
 
   def write(trace, services)
-    super(trace, services)
-    @spans << trace
-    @services = services
+    @mutex.synchronize do
+      super(trace, services)
+      @spans << trace
+      @services = services
+    end
   end
 
   def spans
-    spans = @spans
-    @spans = []
-    spans.flatten
+    @mutex.synchronize do
+      spans = @spans
+      @spans = []
+      spans.flatten!
+      # sort the spans to avoid test flakiness
+      spans.sort! do |a, b|
```

Comment:
I hesitated a lot before introducing this. Because in many cases it puts your spans upside down. I considered sorting them based on `start_time` first. But then in multtithreaded/concurrent context this is not garanteed. And these are the complex cases we want to sort out. So I ended up with that `ORDER BY name, resource, start_time, end_time` plan.